client/client-multi/client-c/src/kaa/kaa_common_schema.h

```diff
@@ -38,7 +38,7 @@ extern "C" {
 
 
 typedef void (*serialize_fn)(avro_writer_t writer, void *data);
-typedef void *(*deserialize_fn)(void);
+typedef void *(*deserialize_fn)(); // NOTE: Do not add void in the parameter list.
```

Comment:
deserialize_fn is a something like a parent class for deserialize_wo_ctx_fn and deserialize_w_ctx_fn. The signature of these typedefs of pointer to functions is different so we can't use void*

Maybe it isn't a good practice to use this kind of hack, but now to make it to more clear a lot of code need to be re-implemented in Avro Gen C.